### PR TITLE
3x tweaks

### PR DESCRIPTION
**Pull request by manolama**

### CODE REVIEW

File: core/src/test/java/net/opentsdb/query/processor/groupby/TestGroupByNumericSummaryParallelIterator.java [hunk 1]
```diff
@@ -92,12 +92,20 @@ public static void beforeClass() {
...
     when(FACTORY.predicate()).thenReturn(p);
     when(FACTORY.tsdb()).thenReturn(TSDB);
+    TSDB.getRegistry().registerFactory(FACTORY);
     
-    GroupByJobPool allocator = FACTORY.new GroupByJobPool();
+    GroupByJobPool allocator = new GroupByJobPool();
+    try {
+      allocator.initialize(TSDB, null).join();
+    } catch (InterruptedException e) {
+      e.printStackTrace();
```
Comment (smrutilal2) [inline core/src/test/java/net/opentsdb/query/processor/groupby/TestGroupByNumericSummaryParallelIterator.java:101]: log error instead

File: core/src/main/java/net/opentsdb/pools/BlockingQueueArrayObjectPool.java [hunk 2]
```diff
@@ -103,8 +103,10 @@ public PooledObject claim(final long time, final ChronoUnit unit) {
   @Override
   public PooledObject claim(final int length) {
     if (length > ((ArrayObjectPoolAllocator) config.allocator()).pooledLength()) {
-      tsdb.getStatsCollector().incrementCounter("objectpool.claim.largeArray", 
+      tsdb.getStatsCollector().incrementCounter("objectpool.claim.lengthTooBig.count", 
           "pool", config.id());
+      tsdb.getStatsCollector().setGauge("objectpool.claim.lengthTooBig.size",
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/pools/BlockingQueueArrayObjectPool.java:108]: This is the metric to track array size.  👍

File: core/src/test/java/net/opentsdb/query/processor/groupby/TestGroupByNumericSummaryParallelIterator.java (hunk 1 above)
Comment (manolama) [reply core/src/test/java/net/opentsdb/query/processor/groupby/TestGroupByNumericSummaryParallelIterator.java:101]: naw just a UT.

File: core/src/main/java/net/opentsdb/pools/BlockingQueueArrayObjectPool.java (hunk 2 above)
Comment (smrutilal2) [reply core/src/main/java/net/opentsdb/pools/BlockingQueueArrayObjectPool.java:108]: The `Gauge` metric will give the count as well. We don't need another `Counter` metric

**Pull request merged**